flink-runtime/src/main/java/org/apache/flink/runtime/state/TaskExecutorStateChangelogStoragesManager.java

```diff
@@ -124,10 +125,11 @@ public void releaseStateChangelogStorageForJob(@Nonnull JobID jobId) {
     }
 
     public void shutdown() {
-        if (closed) {
-            return;
+        synchronized (this) {
```

Comment:
I think we don't need `synchronized` if we are using `AtomicBoolean` here.

Reply:
yes, of course.